tracer/src/Datadog.Trace/AppSec/Rcm/Models/AsmDd/RuleSet.cs

```diff
@@ -0,0 +1,51 @@
+﻿// <copyright file="RuleSet.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System.Collections.Generic;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.AppSec.Rcm.Models.AsmDd;
+
+internal class RuleSet
+{
+    internal string? Version { get; set; }
+
+    internal JToken? Metadata { get; set; }
+
+    [JsonProperty("rules")]
+    internal JToken? Rules { get; set; }
+
+    public JToken? All { get; set; }
+
+    public static RuleSet From(JToken result)
+    {
+        var ruleset = new RuleSet { Version = result["version"]?.ToString(), Metadata = result["metadata"], Rules = result["rules"], All = result };
```

Comment:
this is necessary now as with one waf update with all concatenated products, these 3 keys need all be at the same level with other keys from other products: 
````"rules_override": [...],
   "exclusions": [...],
   "rules_data": [...],
   "version": "2.2",
   "metadata": {
      "rules_version": "1.3.2"
   },
   "rules": [
      {
         "id": "example",
         "name": "example Addresses",
         "tags": {
            "type": "example",
            "category": "example"
         },
         "conditions": []
      },
      // other rules
      ]
   }````